src/app/services/article.service.ts

```diff
@@ -1,25 +1,50 @@
-import { Injectable } from '@angular/core';
-import { Article } from "../interfaces/article";
-import { HttpClient } from "@angular/common/http";
-import { delay, switchMap, from, Observable, of, tap } from "rxjs";
-import { getHardcodedArticles, getHardcodedArticle } from './hardcoded.article';
+import {Injectable} from '@angular/core';
+import {Article} from "../interfaces/article";
+import {from, Observable, of, switchMap, take} from "rxjs";
+import {getHardcodedArticle} from './hardcoded.article';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ArticleService {
-  private articles: Article[] | null = [];
+  private articles: Article[] = [];
   private featuredArticles: string[] = [
-    "stories", 
-    "church-construction-complete", 
-    "is-peter-rock", 
+    "stories",
+    "church-construction-complete",
+    "is-peter-rock",
     "why-not-catholic"
   ];
   private defaultImageURI = "https://t4.ftcdn.net/jpg/02/97/78/03/360_F_297780357_pK8VCA7wctbTFusAGiCfcoxbJLRwC9Bs.jpg";
-  constructor(private http: HttpClient) {
+
+  constructor() {
   }
 
-  loadArticles = async () => {
+  public readOne(articleId: string | null): Observable<Article> {
+    return this.readAll().pipe(take(1),
+      switchMap(() => {
+        const article = this.articles.find(article => article.articleId === articleId);
```

Comment:
Let's fetch the article from the database. I think a literal read `ALL` is not sustainable. I think it can work with 20, maybe 50 articles, but certainly not 200 or more.

Reply:
So, this function actually utilized what you had previous:

- It executes readAll() which will actually only fetch articles if and only if the service's _this.articles_ is NOT already populated. Hence, if it already is populated, it will immediately return it's cache, as it should so we don't fetch more than once.
- Secondly, you included the concept of _featuredArticles_ and fetched _only_ featured articles in your implementation of _loadArticles()_. _readAll()_ is actually utilizing your implementation of _loadArticles()_.